### PR TITLE
Remove array fields

### PR DESCRIPTION
**Pull request by sdc50**
Replaces ArrayFields with JSONFields. 
Update documentation on app testing.
Fix annoyances with app settings:

- Ensure that new settings are synced when syncing an app with the database
- Ensure the obsolete settings are removed when syncing an app
- Prompt user if app should be removed from database when there is an error when loading in `DEBUG` mode.

### CODE REVIEW

File: tethys_apps/base/app_base.py [hunk 1]
```diff
@@ -1634,6 +1635,33 @@ def persistent_store_exists(cls, name):
...
+        # wps settings
+        db_app.sync_settings(
+            self.web_processing_service_settings(), db_app.wps_services_settings
+        )
+        # persistent store settings
+        db_app.sync_settings(
+            self.persistent_store_settings(),
+            db_app.persistent_store_connection_settings
+            | db_app.persistent_store_database_settings,
+        )
```
Comment (swainn) [inline tethys_apps/base/app_base.py:1659]: I'm unfamiliar with the `|` syntax used here. What is the intended effect?

File: tethys_apps/base/app_base.py [hunk 2]
```diff
@@ -1731,11 +1733,28 @@ def remove_from_db(self):
...
+                    proceed = True
+                if response.lower() == "n":
+                    proceed = False
+
+        if proceed:
+            try:
+                # Attempt to delete the object
+                TethysApp.objects.filter(package__exact=self.package).delete()
+            except Exception as e:
+                tethys_log.error(e)
```
Comment (swainn) [inline tethys_apps/base/app_base.py:1757]: Finally! Thank you!

File: tethys_cli/scaffold_templates/app_templates/default/tethysapp/+project+/tests/tests.py_tmpl [hunk 3]
```diff
@@ -28,26 +23,13 @@ To run any tests:
...
-            Test command: "tethys test -f tethys_apps.tethysapp.{{project}}.tests.tests.{{class_name}}TestCase"
-
-        To run only the test_if_tethys_platform_is_great function in the {{class_name}}TestCase class:
-            Test command: "tethys test -f tethys_apps.tethysapp.{{project}}.tests.tests.{{class_name}}TestCase.test_if_tethys_platform_is_great"
 
 To learn more about writing tests, see:
-    https://docs.djangoproject.com/en/1.9/topics/testing/overview/#writing-tests
-    https://docs.python.org/2.7/library/unittest.html#module-unittest
-"""
+    https://docs.tethysplatform.org/en/stable/tethys_sdk/testing.html
```
Comment (swainn) [inline tethys_cli/scaffold_templates/app_templates/default/tethysapp/+project+/tests/tests.py_tmpl:33]: Seems like there is a missing set of `"""` here.

File: tethys_cli/scaffold_templates/app_templates/react/tethysapp/+project+/tests/tests.py_tmpl [hunk 4]
```diff
@@ -28,26 +23,13 @@ To run any tests:
...
-            Test command: "tethys test -f tethys_apps.tethysapp.{{project}}.tests.tests.{{class_name}}TestCase"
-
-        To run only the test_if_tethys_platform_is_great function in the {{class_name}}TestCase class:
-            Test command: "tethys test -f tethys_apps.tethysapp.{{project}}.tests.tests.{{class_name}}TestCase.test_if_tethys_platform_is_great"
 
 To learn more about writing tests, see:
-    https://docs.djangoproject.com/en/1.9/topics/testing/overview/#writing-tests
-    https://docs.python.org/2.7/library/unittest.html#module-unittest
-"""
+    https://docs.tethysplatform.org/en/stable/tethys_sdk/testing.html
```
Comment (swainn) [inline tethys_cli/scaffold_templates/app_templates/react/tethysapp/+project+/tests/tests.py_tmpl:34]: Missing `"""`?

**Review by swainn — CHANGES_REQUESTED** [review]

File: tethys_apps/base/app_base.py (hunk 1 above)
Comment (sdc50) [reply tethys_apps/base/app_base.py:1659]: `db_app.persistent_store_connection_settings` and `db_app.persistent_store_database_settings` are both `QuerySets`. The `|` operator is how to join both sets.

**Comment by sdc50** [comment]
@swainn thanks for reviewing and catching a few things. I've fixed up those issues that you mentioned, so I think this is ready to go now.

**Pull request merged**